src/Avalonia.Base/StyledProperty.cs

```diff
@@ -220,6 +220,27 @@ internal override void RouteClearValue(AvaloniaObject o)
             }
         }
 
+        [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = TrimmingMessages.ImplicitTypeConvertionSupressWarningMessage)]
+        internal override void RouteSetCurrentValue(AvaloniaObject target, object? value)
```

Comment:
This method is identical to `RouteSetValue` save for the one line that calls `SetCurrentValue`. The logic should to be extracted to a shared method that returns true if the caller should proceed to set a new value.

Reply:
Yeah I noticed you'd refactored it into a separate method; I tend to follow the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming) so didn't do it, but it's not a hard-and-fast rule.

Reply:
Refactored this into a separate method. I'm not entirely happy with it as it doesn't meet the rule of three or "methods should be easy to name" rule, but I guess it's not worse than before. I went through a few names for the method but wasn't happy with any of them. Settled on the one that at least reads best.